File: arena/src/main/java/org/zenika/arena/entity/BattleEntity.java
package org.zenika.arena.entity;

import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.TypeAlias;
import org.springframework.data.mongodb.core.mapping.Document;
import org.zenika.core.Battle;
import org.zenika.core.FightingPokemon;

@Document(collection="battle")
@TypeAlias("battle")
public class BattleEntity extends Battle {

    @Id
    private String id;

    public BattleEntity() {
        super();
    }

    public BattleEntity(Battle battle) {
        super(battle.getPokemonA(), battle.getPokemonB());
        this.setRound(battle.getRound());
        this.setWinner(battle.getWinner());
    }


    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }
}

File: arena/src/main/kotlin/org/zenika/arena/entity/BattleEntity.kt
package org.zenika.arena.entity

import org.springframework.data.annotation.Id
import org.springframework.data.annotation.TypeAlias
import org.springframework.data.mongodb.core.mapping.Document
import org.zenika.core.Battle
import org.zenika.core.FightingPokemon
import org.zenika.core.Pokemon

@Document(collection="battle")
@TypeAlias("battle")
class BattleEntity(pokemonA: FightingPokemon, pokemonB: FightingPokemon): Battle(pokemonA, pokemonB) {

    constructor(battle: Battle) : this(battle.pokemonA, battle.pokemonB) {
        this.apply {
            winner = battle.winner
            round = battle.round
        }
    }

    @Id
    var id: String? = null
}
